### ISSUE TITLE
[Bug] Chapters skipping

### ISSUE THREAD

**Issue by morcefaster** [inorichi/tachiyomi#2727]
While #2718 fixed perpetual loading on chapters, it appears that from time to time a chapter gets skipped entirely now (after a very short progress bar).

This might point to what the origin of the race condition was previously. Creating this issue to track the investigation.

EDIT1: The chapter loads, but skips to the last page. Seek bar at the bottom shows last page of previous chapter. It seems that the wrong pages are loaded into the chapter.

**Comment by morcefaster** [comment]
Got'em. I was right about the race condition.

```2020-03-18 00:57:31.830 Loading pages for 8.1
2020-03-18 00:57:31.836 Recycling chapter Vol.3 Chapter 8.1 (8.1)
2020-03-18 00:57:31.853 Loaded chapter 8.1 with 10 pages
```
It seems that for some reason chapter got recycled after pages started loading, which leads to chapter having state=Loaded with no page loader (leading to permanent loading issue, #2715)

The quickfix which made the chapter reload again if the loader is not present loads new Page objects, confusing the hell out of @CarlosEsco's DirectionalViewPager.dataSetChanged() which issues a repopulation and sets current item to what is usually the last page, leading to "skipping" the chapter.

Now going to investigate the reason behind the weird mid-task ref drop.

**Comment by morcefaster** [comment]
Fixed with #2731
